### PR TITLE
Dashboard dev alr

### PR DESCRIPTION
**Pull request by reid-andrew**
## Description
Write and pass test for displaying user info on dashboard.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which maintains existing functionality)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Notes
Created the following objects:
- User Model
- Dashboard Facade
- FlatService
-Dashboard Controller
-Dashboard Index View

## RSpec Results
```
Finished in 0.54152 seconds (files took 3.05 seconds to load)
2 examples, 0 failures
```

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,7 @@
 class ApplicationController < ActionController::Base
+  helper_method :current_user
+
+  def current_user
+    @user ||= User.find(session[:user_id]) if session[:user_id]
+  end
 end
```
Comment (reid-andrew) [inline app/controllers/application_controller.rb:7]: Need to add sessions for real. This method is currently getting stubbed & not actually triggering the requirements to have a session.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,2 +1,7 @@
 class User < ApplicationRecord
+  validates_presence_of :email
+
+  def flat_key
+    ENV['FLAT_KEY']
+  end
 end
```
Comment (reid-andrew) [inline app/models/user.rb:7]: Need to replace hardcoding with OAuth token reference once that functionality is built out.

File: app/services/flat_service.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class FlatService
+  class << self
+    def get_user(current_user)
+      resp = conn(current_user).get('/v2/me')
+      # require "pry"; binding.pry
+      get_json(resp)
```
Comment (reid-andrew) [inline app/services/flat_service.rb:6]: Sorry - my b leaving pry in there.

File: app/facades/dashboard_facade.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+class DashboardFacade
+  attr_reader :user_info, :scores
+
+  def initialize(current_user)
+    @user_info = FlatService.get_user(current_user)
```
Comment (zachholcomb) [inline app/facades/dashboard_facade.rb:5]: We'll have a better idea of this later, wonder if we'll want to have individual score objects down the road, not sure yet though

File: app/views/users/dashboard/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+
+<p>User Name: <%= @user.user_info[:username] %></p>
+<p>Number of Scores: <%= @user.user_info[:ownedPublicScoresCount] %></p>
```
Comment (zachholcomb) [inline app/views/users/dashboard/index.html.erb:3]: also same idea here, wondering if a user_info object would be cleaner, might not for this case though

**Pull request merged**